### PR TITLE
fix: whitespace parsing and explanation for ContainsEvaluator

### LINKED ISSUE #11101: contains evaluator doesn't really have a useful explanation, doesn't seem to actaully work correctly
create a contains evaluator that's supposed to match on spices for recipes. It fails to go 1

Image

### PR DESCRIPTION
**Pull request by anticorrelator**
resolves #11101

---

> [!NOTE]
> **Low Risk**
> Behavior changes are narrowly scoped to the `contains` built-in evaluator and its user-facing explanations, with added unit coverage to prevent regressions.
> 
> **Overview**
> Fixes `ContainsEvaluator` word parsing so whitespace-only entries and trailing commas are ignored and an empty word list returns `False` (preventing vacuous/false-positive matches), while adding `require_all` support to the example code snippets.
> 
> Updates `ContainsEvaluator`’s description and overhauls explanations to include a clean word list, a truncated text preview, and explicit matched/missing details; tests are added/updated to cover trailing-comma/empty-input behavior and the new explanation phrasing (including preview mutation assertions).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7f231402234a6ef8f192accd068ffe7fce7a9094. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/phoenix/server/api/evaluators.py [hunk 1]
```diff
@@ -1285,14 +1289,44 @@ async def _evaluate(
...
+                            matched_list = ", ".join(found_words)
+                            explanation = (
+                                f"One or more of the words [{word_list}]"
+                                f" were found in the text:"
+                                f" '{truncated_text}'."
+                                f" Matched: {matched_list}"
+                            )
+                        else:
+                            explanation = (
+                                f"One or more of the words [{word_list}]"
```
Comment (yfrigui2) [inline src/phoenix/server/api/evaluators.py]: nit for clarity
```suggestion
                                f"None of the words [{word_list}]"
```
